src/server/frontend_wayland/wl_surface_role.h

```diff
@@ -49,13 +49,13 @@ struct WlSurfaceState;
 class WlSurfaceRole
 {
 public:
-    virtual bool synchronized() const { return false; }
+    virtual bool synchronized() const = 0;
```

Comment:
Only subsurfaces have the option to be synchronized. The dafault implementation makes it clear that you don't have to worry about this unless you have a good reason to.

Reply:
Also, default reduces amount of boilerplate code in subclasses (makes them more readable)

Reply:
I think it surprising to see an inline implementation of a virtual function, especially in what is otherwise an interface.

But I can change it back as it isn't needed for this PR.